src/vendorpromo/models.py

```diff
@@ -72,8 +77,11 @@ class Meta:
         verbose_name_plural = "Affiliates"
 
     def clean(self):
-        if (self.customer_profile is None) and (self.full_name is None and self.email is None and self.company is None):
-            raise ValidationError(_("You at least need to assign a Customer Profile of enter a Full Name, Email or Company for the Affiliate"))
+        if (self.customer_profile is None) and (self.contact_name is None and self.email is None and self.company is None):
+            raise ValidationError(_("You at least need to assign a Customer Profile or enter a Full Name, Email or Company for the Affiliate"))
+        
+        if self.customer_profile is not None and Affiliate.objects.filter(customer_profile=self.customer_profile).exists():
```

Comment:
should this check be site specific? what if they want to be an affiliate for multiple sites?

Reply:
never mind, forgot that customer profiles were site specific